contracts/sfc/SFC.sol

```diff
@@ -419,6 +429,22 @@ contract SFC is OwnableUpgradeable, UUPSUpgradeable, Version {
         }
     }
 
+    /// Resolve failed treasury transfers and send the unresolved fees to the treasury address.
+    function resolveTreasuryFees() external nonReentrant {
+        if (treasuryAddress == address(0)) {
+            revert TreasuryNotSet();
+        }
+        if (unresolvedTreasuryFees == 0) {
+            revert NoUnresolvedTreasuryFees();
+        }
+        (bool success, ) = treasuryAddress.call{value: unresolvedTreasuryFees, gas: 1000000}("");
+        if (!success) {
+            revert TransferFailed();
+        }
+        emit TreasuryFeesResolved(unresolvedTreasuryFees);
+        unresolvedTreasuryFees = 0;
```

Comment:
To follow Checks-Effects-Interactions Pattern (SWC-107), following order should be used:
1. checks
2. effects - state modification (setting unresolvedTreasuryFees to zero)
3. interactions - other contract calls (sending the money)

See https://docs.soliditylang.org/en/latest/security-considerations.html#use-the-checks-effects-interactions-pattern

Reply:
It's using re-entracy guard

Reply:
Please follow the convention to avoid unnecessary audit notes.